code/modules/mob/living/carbon/human/species/diona.dm

```diff
@@ -86,7 +86,7 @@
 
 /datum/species/diona/handle_life(mob/living/carbon/human/H)
 	var/light_amount = 0 //how much light there is in the place, affects receiving nutrition and healing
-	var/is_vamp = H.mind?.vampire != null
+	var/is_vamp = H.mind?.has_antag_datum(/datum/antagonist/vampire) != null
```

Comment:
```suggestion
	var/is_vamp = !!H.mind?.has_antag_datum(/datum/antagonist/vampire)
```
`has_antag_datum` can return FALSE. Which is not null. `!!` will make it a boolean value. Anything -> TRUE, null/FALSE/ etc -> FALSE